credentials/src/test/kotlin/web5/sdk/credentials/VerifiablePresentationTest.kt

```diff
@@ -0,0 +1,246 @@
+import com.nimbusds.jose.JWSHeader
+import com.nimbusds.jose.JWSSigner
+import com.nimbusds.jose.crypto.Ed25519Signer
+import com.nimbusds.jose.jwk.Curve
+import com.nimbusds.jose.jwk.gen.OctetKeyPairGenerator
+import com.nimbusds.jwt.JWTClaimsSet
+import com.nimbusds.jwt.SignedJWT
+import org.junit.jupiter.api.Assertions.assertThrows
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertDoesNotThrow
+import web5.sdk.credentials.model.InputDescriptorMapping
+import web5.sdk.credentials.model.PresentationSubmission
+import web5.sdk.crypto.InMemoryKeyManager
+import web5.sdk.dids.methods.ion.CreateDidIonOptions
+import web5.sdk.dids.methods.ion.DidIon
+import web5.sdk.dids.methods.ion.JsonWebKey2020VerificationMethod
+import web5.sdk.dids.methods.key.DidKey
+import java.security.SignatureException
+import java.text.ParseException
+import java.util.UUID
+import kotlin.test.assertEquals
+import kotlin.test.assertNotNull
+
+class VerifiablePresentationTest {
+
+  @Test
+  fun `create simple vp`() {
+    val vcJwts: Iterable<String> = listOf("vcjwt1")
+
+    val keyManager = InMemoryKeyManager()
+    val holderDid = DidKey.create(keyManager)
+
+    val vp = VerifiablePresentation.create(
+      vcJwts = vcJwts,
+      holder = holderDid.uri
+    )
+
+    assertNotNull(vp, "VerifiablePresentation should not be null")
+    assertEquals(holderDid.uri, vp.holder, "holder should match")
+    assertEquals(vcJwts, vp.verifiableCredential, "vcJwts should match")
+  }
+
+  @Test
+  fun `create vp with presentation submission`() {
+    val keyManager = InMemoryKeyManager()
+    val holderDid = DidKey.create(keyManager)
+
+    val vcJwts: Iterable<String> = listOf("vcjwt1", "vcjwt2")
+
+    val presentationSubmission = PresentationSubmission(
+      id = "presentationSubmissionId",
+      definitionId = "definitionId",
+      descriptorMap = listOf(
+        InputDescriptorMapping(
+          id = "descriptorId",
+          format = "format",
+          path = "path"
+        )
+      )
+    )
+
+    val vp = VerifiablePresentation.create(
+      vcJwts = vcJwts,
+      holder = holderDid.uri,
+      additionalData = mapOf("presentation_submission" to presentationSubmission)
+    )
+
+    val vpDataModelMap = vp.vpDataModel.toMap()
+    val mappedPresentationSubmission = vpDataModelMap["presentation_submission"] as? PresentationSubmission
+
+    assertNotNull(mappedPresentationSubmission, "Mapped PresentationSubmission should not be null")
+    assertEquals(presentationSubmission, mappedPresentationSubmission, "PresentationSubmission should match")
+  }
+
+
+  @Test
+  fun `simple sign works`() {
+    val keyManager = InMemoryKeyManager()
+    val holderDid = DidKey.create(keyManager)
+
+    val vp = VerifiablePresentation.create(
+      vcJwts = listOf("vcjwt1", "vcjwt2"),
+      holder = holderDid.uri
+    )
+
+    val vpJwt = vp.sign(holderDid)
+    assertNotNull(vpJwt)
+  }
+
+  @Test
+  fun `presentationSubmission sign works`() {
```

Comment:
`works` is pretty vague. Should this test name clarify which behavior is being tested?